### DISCUSSION TITLE
Installing R and RStudio using homebrew does not work together

### DISCUSSION THREAD

**Discussion by fhollenbach** [Homebrew/discussions#1508] [Everyday usage] [answered]
Hi, 

I am on a new M1 MacBook Air with OS 11.3.1 (BigSur). Yesterday I reinstalled R and RStudio using homebrew and the two installs do not seem to work together. 
I first ran: 
'brew install R', which successfully installed R4.1.0 in /opt/homebrew/bin/R
I then ran:
brew install --cask RStudio, which successfully installed RStudio. 
But when I try to open RStudio, it fails with the error: "Unable to locate R binary by scanning standard locations". 
If I follow the advice here: https://support.rstudio.com/hc/en-us/articles/200486138-Changing-R-versions-for-RStudio-desktop and change the path to /opt/homebrew/bin/R, I get the error screen that is attached here. 
Running R in Terminal or via Emacs works just fine. 

Do I need to install a separate R version just for RStudio? Are there sys links that are missing? Running brew doctor results in  "Your system is ready to brew."

Any help would be greatly appreciated! Thank you

<img width="1186" alt="Screen Shot 2021-05-21 at 9 02 08 AM" src="https://user-images.githubusercontent.com/1372634/119150001-a88eae00-ba13-11eb-9351-6af101cdb484.png">
<img width="1181" alt="Screen Shot 2021-05-21 at 9 02 28 AM" src="https://user-images.githubusercontent.com/1372634/119150011-aa587180-ba13-11eb-959e-3863531365c0.png">

**Comment by gromgit** [comment, marked as answer]
I don't use R myself, but `brew info --cask rstudio` explicitly says (and you would've been shown this during `brew install --cask rstudio`):
```
==> Caveats
rstudio depends on R. The R Project provides official binaries:

  brew install --cask r

Alternatively, the Homebrew-compiled version of R omits the GUI app:

  brew install r
```
Based on the "wrong architecture" error that RStudio threw, my first guess would be that the RStudio app is still Intel-based, and would therefore not be compatible with the M1-based `r` formula. Sure enough:
```
$ file /Applications/RStudio.app/Contents/MacOS/RStudio 
/Applications/RStudio.app/Contents/MacOS/RStudio: Mach-O 64-bit executable x86_64
```
Try doing what the `rstudio` caveat recommends, but clean up the M1 `r` first:
```
brew uninstall r
brew install --cask r
```

**Reply by fhollenbach** [reply]
Thanks! 
I did see the caveat when installing Rstudio, however, I did not realize that the caveat implied it would not work with the homebrew-compiled version.

**Reply by carlocab** [reply]
That's because it does work when you're running it on Intel (or via Rosetta).

**Reply by fhollenbach** [reply]
Makes sense, thanks!

**Comment by fhollenbach** [comment]
My mistake was that I didn't realize the brew install R command now installs R build for silicon macs but RStudio only runs on the version via Rosetta. 
I now installed one of the daily builds of RStudio, which has silicon support (if interested, see link in here: https://community.rstudio.com/t/update-on-r-rstudio-support-for-apple-silicon-macs/95582) and it works with the R version installed via 'brew install R'

Thank you all.

**Reply by carlocab** [reply]
You can probably install that with one of
```
brew install --cask rstudio-daily
```
or
```
brew install --cask rstudio-preview
```
You may need to do `brew tap homebrew/cask-versions` first.

**Reply by fhollenbach** [reply]
Thank you!
